controllers/flinkcluster/flinkcluster_reconciler.go

```diff
@@ -834,6 +854,20 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 	return ctrl.Result{}, nil
 }
 
+func (reconciler *ClusterReconciler) isSameJobSubmitter(observedJob *batchv1.Job, desiredJob *batchv1.Job) (bool, error) {
```

Comment:
@nickcaballero can you expand on why we need the create dry-run here and not just comparing with the desiredJob?

Reply:
Just to simplify the comparison.